src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -6190,7 +6190,11 @@ static QVariant fcnArrayInsert( const QVariantList &values, const QgsExpressionC
 static QVariant fcnArrayRemoveAt( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
 {
   QVariantList list = QgsExpressionUtils::getListValue( values.at( 0 ), parent );
-  list.removeAt( QgsExpressionUtils::getNativeIntValue( values.at( 1 ), parent ) );
+  int position = QgsExpressionUtils::getNativeIntValue( values.at( 1 ), parent );
+  if ( position < 0 && ( list.length() + position ) >= 0 )
+    position = position + list.length();
+  if ( position < list.length() )
```

Comment:
What if 
list is [1, 2, 3] 
position is -100
?
will you execute list.removeAt( -100 );
?
I think you need to also check for position >= 0

Reply:
You're right I've overlooked that case, thanks!